mc/util.go

```diff
@@ -59,20 +59,77 @@ func ParseHandleId(str string) (empty p2p_pstore.PeerInfo, err error) {
 	return p2p_pstore.PeerInfo{ID: pid}, nil
 }
 
-func NewHost(id PeerIdentity, addrs ...multiaddr.Multiaddr) (p2p_host.Host, error) {
+// re-export this option to avoid basic host interface leakage
+const NATPortMap = p2p_bhost.NATPortMap
+
+func NewHost(id PeerIdentity, laddr multiaddr.Multiaddr, opts ...interface{}) (p2p_host.Host, error) {
 	pstore := p2p_pstore.NewPeerstore()
 	pstore.AddPrivKey(id.ID, id.PrivKey)
 	pstore.AddPubKey(id.ID, id.PrivKey.GetPublic())
 
 	netw, err := p2p_swarm.NewNetwork(
 		context.Background(),
-		addrs,
+		[]multiaddr.Multiaddr{laddr},
 		id.ID,
 		pstore,
 		p2p_metrics.NewBandwidthCounter())
 	if err != nil {
 		return nil, err
 	}
 
-	return p2p_bhost.New(netw), nil
+	return p2p_bhost.New(netw, opts...), nil
+}
+
+// multiaddr juggling
+func isAddrSubnet(addr multiaddr.Multiaddr, prefix []string) bool {
+	ip, err := addr.ValueForProtocol(multiaddr.P_IP4)
+	if err != nil {
+		return false
+	}
+
+	for _, pre := range prefix {
+		if strings.HasPrefix(ip, pre) {
+			return true
+		}
+	}
+
+	return false
+}
+
+var (
+	localhostSubnet  = []string{"127."}
+	linkLocalSubnet  = []string{"169.254."}
+	privateSubnet    = []string{"10.", "172.16.", "192.168."}
+	unroutableSubnet = []string{"0.", "127.", "169.254."}
+	internalSubnet   = []string{"0.", "127.", "169.254.", "10.", "172.16.", "192.168."}
```

Comment:
that's a little tricky, because I can't do the straight forward prefix thing (it's /10).